client/post-editor/media-modal/detail/detail-fields.jsx

```diff
@@ -89,14 +88,14 @@ class EditorMediaModalDetailFields extends Component {
 
 		// Save changes immediately or after a delay
 		if ( saveImmediately ) {
-			this.saveChange( siteId, modifiedChanges );
+			this.saveChange( siteId, itemId, modifiedChanges );
 		} else {
-			this.delayedSaveChange( siteId, modifiedChanges );
+			this.delayedSaveChange( siteId, itemId, modifiedChanges );
 		}
 	}
 
-	saveChange( siteId, modifiedChanges ) {
-		this.props.updateMedia( siteId, modifiedChanges );
+	saveChange( siteId, mediaId, modifiedChanges ) {
+		this.props.updateMedia( siteId, mediaId, modifiedChanges );
```

Comment:
Before this PR, the `mediaId` was stored as `this.state.modifiedChanges.ID`. Now you can update `setFieldByName` to no longer add the `ID` field, as it's no longer used.

Reply:
I added https://github.com/Automattic/wp-calypso/pull/58604/commits/ea85db914aa5ffcba3ed3a7ab88fa662c2c184d9 which includes your suggestions and a few more changes regarding state.